core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -732,6 +732,13 @@ class DAGScheduler(
   }
 
   /**
+   * Kill a given task. It will be retried.
+   */
+  def killTask(taskId: Long, reason: String): Unit = {
+    taskScheduler.killTask(taskId, true, reason)
+  }
```

Comment:
always interruptThread ? Thread interrupt'ion has other implications, particularly on lower stack libraries.
See https://github.com/apache/spark/pull/17113#issuecomment-284501759 for example